LongWarOfTheChosen/Src/LW_Overhaul/Classes/X2EventListener_Headquarters.uc

```diff
@@ -78,6 +92,65 @@ static protected function int GetListenerPriority()
 	return default.LISTENER_PRIORITY != -1 ? default.LISTENER_PRIORITY : class'XComGameState_LWListenerManager'.default.DEFAULT_LISTENER_PRIORITY;
 }
 
+// KDM : Listen for navigation help updates within UIArmory_WeaponUpgrade
+static function EventListenerReturn OnWeaponUpgradeNavHelpUpdated(
+	Object EventData,
+	Object EventSource,
+	XComGameState NewGameState,
+	Name InEventID,
+	Object CallbackData)
+{
+	local bool AllowWeaponStripping, IsUpgradesListSelected;
+	local UIArmory_WeaponUpgrade WeaponUpgradeScreen;
+	local UIList UpgradesList;
+	local UINavigationHelp NavHelp;
+	local UIPanel UpgradesListContainer;
+	local XComGameState_LWListenerManager ListenerManager;
+	
+	WeaponUpgradeScreen = UIArmory_WeaponUpgrade(EventSource);
+	NavHelp = UINavigationHelp(EventData);
+
+	if (NavHelp == none)
+	{
+		`LWTrace("OnWeaponUpgradeNavHelpUpdated event did not have UINavigationHelp as its data");
+		return ELR_NoInterrupt;
+	}
+
+	if ((WeaponUpgradeScreen == none) || (UIArmory_WeaponTrait(WeaponUpgradeScreen) != none))
+	{
+		`LWTrace("OnWeaponUpgradeNavHelpUpdated event did not have UIArmory_WeaponUpgrade as its source");
+		return ELR_NoInterrupt;
+	}
+
+	ListenerManager = class'XComGameState_LWListenerManager'.static.GetListenerManager();
+	UpgradesListContainer = WeaponUpgradeScreen.UpgradesListContainer;
+	UpgradesList = WeaponUpgradeScreen.UpgradesList;
+
+	IsUpgradesListSelected = ((WeaponUpgradeScreen.Navigator.GetSelected() == UpgradesListContainer) && (UpgradesListContainer.Navigator.GetSelected() == UpgradesList));
+	// KDM : Don't allow weapon stripping if either the 1.] upgrade slot list is open 2.] colour selector is open.
+	AllowWeaponStripping = (!(IsUpgradesListSelected || (WeaponUpgradeScreen.ColorSelector != none)));
+	
+	if (AllowWeaponStripping)
+	{
+		// KDM : 'Strip Weapon Upgrades' is a CustomizeList list item for mouse & keyboard users, so it doesn't need to be added to the
+		// navigation help system.
+		if (`ISCONTROLLERACTIVE)
+		{
+			// KDM : Right stick click corresponds to 'Strip Weapon Upgrades'
+			NavHelp.AddRightHelp(CAPS(class'UIScreenListener_ArmoryWeaponUpgrade_LW'.default.strStripWeaponUpgradesButton), 
+				class'UIUtilities_Input'.const.ICON_RSCLICK_R3, , false, 
+				class'UIScreenListener_ArmoryWeaponUpgrade_LW'.default.strStripWeaponUpgradesTooltip);
+		}
+
+		// KDM : Left stick click corresponds to 'Strip All Weapon Upgrades'.
+		NavHelp.AddRightHelp(CAPS(class'UIScreenListener_ArmoryWeaponUpgrade_LW'.default.StripAllWeaponUpgradesStr), 
+			class'UIUtilities_Input'.const.ICON_LSCLICK_L3, ListenerManager.OnStripUpgrades, false,
```

Comment:
We should be moving code out of `XComGameState_LWListenerManager`. In fact, I'm wondering why LW2 required a couple of custom highlander events to insert those buttons. Any ideas, since you've managed to do at least one of them without the event used in `XCGS_LWListenerManager`?